File: app/src/main/java/the/one/demo/ui/bean/HomeItemSection.java
package the.one.demo.ui.bean;

//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

import com.qmuiteam.qmui.widget.section.QMUISection;

import java.util.List;

/**
 * @author The one
 * @date 2019/3/4 0004
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
public class HomeItemSection implements QMUISection.Model<HomeItemSection>{

    public String url;
    public String content;
    public List<String> images;
    public String remark;

    public HomeItemSection(String content, List<String> images, String remark,String url) {
        this.content = content;
        this.images = images;
        this.remark = remark;
        this.url = url;
    }

    @Override
    public HomeItemSection cloneForDiff() {
        return new HomeItemSection(content,images,remark,url);
    }

    @Override
    public boolean isSameItem(HomeItemSection other) {
        return content == other.content || (content != null && content.equals(other.content));
    }

    @Override
    public boolean isSameContent(HomeItemSection other) {
        return true;
    }
}

File: kotlin/src/main/java/the/one/kotlin/model/HomeItemSection.kt
package the.one.kotlin.model

//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

import com.qmuiteam.qmui.widget.section.QMUISection

/**
 * @author The one
 * @date 2019/3/4 0004
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
class HomeItemSection(var content: String?, var images: List<String>?, var remark: String, var url: String) : QMUISection.Model<HomeItemSection> {

    override fun cloneForDiff(): HomeItemSection {
        return HomeItemSection(content, images, remark, url)
    }

    override fun isSameItem(other: HomeItemSection): Boolean {
        return content === other.content || content != null && content == other.content
    }

    override fun isSameContent(other: HomeItemSection): Boolean {
        return true
    }
}
